app/src/main/java/cm/aptoide/pt/database/RoomInstalledPersistence.java

```diff
@@ -0,0 +1,131 @@
+      RoomInstallationMapper installationMapper) {
+    this.installedDao = installedDao;
+    this.roomInstallationPersistence = roomInstallationPersistence;
+    this.installationMapper = installationMapper;
+  }
+
+  public Observable<List<RoomInstalled>> getAllInstalled() {
+    return RxJavaInterop.toV1Observable(installedDao.getAll(), BackpressureStrategy.BUFFER)
+        .flatMap(installs -> filterCompleted(installs))
+        .subscribeOn(Schedulers.io());
+  }
+
+  /**
+   * @return all the entries from this table even not installed apps
+   * if you want only installed apps consider using the install manager
+   */
+  public Observable<List<RoomInstalled>> getAll() {
+    return RxJavaInterop.toV1Observable(installedDao.getAll(), BackpressureStrategy.BUFFER)
+        .subscribeOn(Schedulers.io());
+  }
+
+  public Observable<List<RoomInstalled>> getAllInstalledSorted() {
+    return RxJavaInterop.toV1Observable(installedDao.getAllSortedAsc(), BackpressureStrategy.BUFFER)
+        .flatMap(installs -> filterCompleted(installs))
+        .subscribeOn(Schedulers.io());
+  }
+
+  public Completable remove(String packageName, int versionCode) {
+    return RxJavaInterop.toV1Completable(installedDao.remove(packageName, versionCode))
+        .subscribeOn(Schedulers.io());
+  }
+
+  public Observable<Boolean> isInstalled(String packageName) {
+    return getInstalled(packageName).map(
+        installed -> installed != null && installed.getStatus() == RoomInstalled.STATUS_COMPLETED);
+  }
+
+  public Observable<RoomInstalled> getInstalled(String packageName) {
+    return getInstalledAsList(packageName).map(installedList -> {
+      if (installedList.isEmpty()) {
+        return null;
+      } else {
+        return installedList.get(0);
+      }
+    });
+  }
+
+  public Observable<RoomInstalled> get(String packageName, int versionCode) {
+    return RxJavaInterop.toV1Observable(installedDao.get(packageName, versionCode),
+        BackpressureStrategy.BUFFER)
+        .subscribeOn(Schedulers.io());
+  }
+
+  public Observable<List<RoomInstalled>> getAsList(String packageName, int versionCode) {
+    return RxJavaInterop.toV1Observable(installedDao.getAsList(packageName, versionCode),
+        BackpressureStrategy.BUFFER)
+        .onErrorReturn(throwable -> new ArrayList<>())
+        .subscribeOn(Schedulers.io());
+  }
+
+  public Completable insert(RoomInstalled installed) {
+    return Completable.fromAction(() -> installedDao.insert(installed))
+        .andThen(roomInstallationPersistence.insert(installationMapper.map(installed)))
+        .subscribeOn(Schedulers.io());
+  }
+
+  public Observable<List<RoomInstalled>> getAllAsList(String packageName) {
+    return RxJavaInterop.toV1Observable(installedDao.getAsListByPackageName(packageName),
+        BackpressureStrategy.BUFFER)
+        .subscribeOn(Schedulers.io());
+  }
+
+  public Completable clearAndAddAll(List<RoomInstalled> list) {
```

Comment:
While not a real issue, I think this could by renamed to replaceAllBy. My suggestion